GH-21 Meetup5 Improve Todo cell UI

ivancr opened the pull request:
- Added completion date property to Item Model
- Added an icon for complete and incomplete state.
- Added label to display completed date
- Setup labels layout to vertical when dynamic type size is large.
- Removed autoresizing masks form xibs
- Cleaned up code and added pragma marks

ivancr committed:
GH-21 Meetup 5 Improve Todo Cell UI:
- Added completion date property to Item Model
- Added an icon for complete and incomplete state.
- Added label to display completed date
- Setup labels layout to vertical when dynamic type size is large.
- Removed autoresizing masks form xibs
- Cleaned up code and added pragma marks

ivancr committed:
GH-21

ivancr committed:
Merge branch 'master' of https://github.com/chuva-io/TodoList into GH-21-Meetup5-Improve-Todo-Cell-UI

# Conflicts:
#	TodoList/View Controllers/TodoListViewController.swift

ivancr commented:
[GH-20 PR](https://github.com/chuva-io/TodoList/pull/25) needs to be merged before we can merge this one.

chuva-io commented:
> [GH-20 PR](https://github.com/chuva-io/TodoList/pull/25) needs to be merged before we can merge this one.

Next time submit the pull request on the branch it depends on. You're requesting to merge it into master.

stemberamichal commented on TodoList/Models/Item.swift:
```diff
@@ -2,10 +2,12 @@ import Foundation
 struct Item: Codable {
     let title: String
+    var completionDate: Date?
     var isComplete: Bool
```
I would expect `isComplete` to be calculated property of `completionDate`.
This can introduce inconsistency between those two.

ivancr replied:
That's a great point @stemberamichal. I'll change that when I get a chance. Thanks!!

ivancr replied:
I created an issue [(GH-30)](https://github.com/chuva-io/TodoList/issues/30)  to deal with this. I don't want to introduce it in this PR that is focused on other topic.

stemberamichal commented on TodoList/View Controllers/TodoListViewController.swift:
```diff
@@ -27,10 +33,7 @@ class TodoListViewController: UITableViewController {
-        tableView.estimatedRowHeight = 44
-        tableView.rowHeight = UITableViewAutomaticDimension
-        tableView.estimatedSectionFooterHeight = 80
-        tableView.sectionFooterHeight = UITableViewAutomaticDimension
```
How did you replace `UITableViewAutomaticDimension`? 
I have looked through the changes and could not find it.

ivancr replied:
`UITableViewAutomaticDimension` is set by default when done programmatically.

https://developer.apple.com/documentation/uikit/uitableview/1614852-rowheight
> The default value of rowHeight is automaticDimension. Note that if you create a self-sizing cell in Interface Builder, the default row height is changed to the value set in Interface Builder

stemberamichal commented on TodoList/View Controllers/TodoListViewController.swift:
```diff
@@ -27,10 +33,7 @@ class TodoListViewController: UITableViewController {
-        tableView.rowHeight = UITableViewAutomaticDimension
-        tableView.estimatedSectionFooterHeight = 80
-        tableView.sectionFooterHeight = UITableViewAutomaticDimension
+        tableView.keyboardDismissMode = .interactive
```
That's so cool. 
Have been using `UIScrollViewDelegate` to dismiss keyboard.

ivancr replied:
Yeah, I love finding these tricks. I try to find the simplest way that uses the least code possible.

ivancr committed:
GH-20 Meetup5 Empty state (#25)

* GH-21 Changes:
- Moved TableView to child VC.
- Handled Keyboard
- Added Empty state
- Fixed textField placeholder Dynamic Type support
- Added Extensions to add and remove child VC's, add subview with fill constraints
- Made Navbar opaque.

* GH-20 Empty state improvements

Commit:
Merge branch 'master' of https://github.com/chuva-io/TodoList into GH-21-Meetup5-Improve-Todo-Cell-UI

# Conflicts:
#	TodoList/AppDelegate.swift
